File: 백준/Silver/2559. 수열/수열.py
import sys

input = sys.stdin.readline

n,k = map(int,input().split(' '))
numbers = list(map(int,input().split(' ')))

current = 0
for i in range(0,k):
  current += numbers[i]

answer = current

for i in range(1,n-k+1):
  current = current - numbers[i-1] + numbers[i+k-1]
  answer = max(answer,current)

print(answer)

File: 백준/Silver/2559. 수열/수열.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const [N,K] = input[0].split(' ').map(Number)
const temperature = input[1].split(' ').map(Number)

let answer = temperature.slice(0,K).reduce((a,b)=>a+b,0)
let test = answer
for(let i=K;i<N;i++){
  test = test - temperature[i-K] + temperature[i]
  answer = Math.max(answer,test)
}

console.log(answer)









